Sources/SwiftKotlinFramework/plugins/IdentifiersTransformPlugin.swift

```diff
@@ -0,0 +1,182 @@
+
+//
+//  IdentifiersTransformPlugin.swift
+//
+//  Created by Tor Langballe on 05/07/2018.
+//
+//
+
+import Foundation
+import Transform
+import AST
+import Source
+
+private func findToken(kind:Token.Kind, val:String, tokens:[Token]) -> Int? {
+    for (i, t) in tokens.enumerated() {
+        if t.kind == kind && t.value == val {
+            return i
+        }
+    }
+    return nil
+}
+
+public class IdentifiersTransformPlugin: TokenTransformPlugin {
+    public var name: String {
+        return "Change identifiers"
+    }
+    
+    public var description: String {
+        return "Changes more functions and casts etc"
+    }
+    
+    public init() {}
+    
+    // this does brute-force conversion of tokens at a general level. Maybe they can be done more in context, but I think it was hard or needed work in AST
+
+    public func transform(tokens: [Token], topDeclaration: TopLevelDeclaration) throws -> [Token] {
+        var newTokens = [Token]()
+        
+        for token in tokens {
+            if let origin = token.origin, let node = token.node {
+                switch token.value {
+                case "&":
+                    if token.kind == .identifier || token.kind == .symbol {
+                        newTokens.append(origin.newToken(.keyword, "and", node))
+                        continue
+                    }
+
+                case "|":
+                    if token.kind == .identifier || token.kind == .symbol {
+                        newTokens.append(origin.newToken(.keyword, "or", node))
+                        continue
+                    }
+
+                case "enumerated":
+                    if token.kind == .identifier {
+                        newTokens.append(origin.newToken(.identifier, "withIndex", node))
+                        continue
+                    }
+                
+//                case "append":
+//                    if token.kind == .identifier {
+//                        if let p = newTokens.last, p.value == "." && p.kind == .delimiter {
+//                            newTokens.removeLast()
+//                            newTokens.append(origin.newToken(.identifier, " += ", node))
+//                        }
+//                        continue
+//                    }
+                    
+                default:
+                    break
+                }
+            }
+            newTokens.append(token)
+        }
+        return newTokens
+    }
+    
+    class func TransformKotlinFunctionDeclarations(_ tokens:[Token]) -> [Token] {
+        // https://kotlinlang.org/docs/reference/operator-overloading.html
+        var newTokens = tokens.filter { $0.kind != .keyword || $0.value != "mutating" }
```

Comment:
The `mutating` keyword was intentionally kept because value types have no counterpart in Kotlin, so I would prefer the dev to check for this error manually. Otherwise, the code could compile, but the resulting execution differ in the Kotlin since it will not create a copy when assigning values to this properties